packages/core-backend/README.md

```diff
@@ -0,0 +1,427 @@
+5. **Parallel Processing**: Transaction and balance updates processed simultaneously - AccountActivityService publishes both transactionUpdated (future) and balanceUpdated events in parallel
+6. **Dynamic Polling**: TokenBalancesController adjusts HTTP polling intervals based on WebSocket connection health (10 min when connected, 20s when disconnected)
+7. **Direct Balance Processing**: Real-time balance updates bypass HTTP polling and update TokenBalancesController state directly
+8. **Connection Resilience**: Automatic reconnection with resubscription to selected account
+9. **Ultra-Simple Error Handling**: Any error anywhere → force reconnection (no nested try-catch)
+
+## API Reference
+
+### BackendWebSocketService
+
+The core WebSocket client providing connection management, authentication, and message routing.
+
+#### Constructor Options
+
+```typescript
+interface BackendWebSocketServiceOptions {
+  messenger: BackendWebSocketServiceMessenger;
+  url: string;
+  timeout?: number;
+  reconnectDelay?: number;
+  maxReconnectDelay?: number;
+  requestTimeout?: number;
+  enableAuthentication?: boolean;
+  enabledCallback?: () => boolean;
+}
+```
+
+#### Methods
+
+- `connect(): Promise<void>` - Establish authenticated WebSocket connection
+- `disconnect(): Promise<void>` - Close WebSocket connection  
+- `subscribe(options: SubscriptionOptions): Promise<SubscriptionResult>` - Subscribe to channels
+- `sendRequest(message: ClientRequestMessage): Promise<ServerResponseMessage>` - Send request/response messages
+- `isChannelSubscribed(channel: string): boolean` - Check subscription status
+- `findSubscriptionsByChannelPrefix(prefix: string): SubscriptionInfo[]` - Find subscriptions by prefix
+- `getConnectionInfo(): WebSocketConnectionInfo` - Get detailed connection state
+
+### AccountActivityService
+
+High-level service for monitoring account activity using WebSocket data.
+
+#### Constructor Options
+
+```typescript
+interface AccountActivityServiceOptions {
+  messenger: AccountActivityServiceMessenger;
+  subscriptionNamespace?: string;
+}
+```
+
+#### Methods
+
+- `subscribeAccounts(subscription: AccountSubscription): Promise<void>` - Subscribe to account activity
+- `unsubscribeAccounts(subscription: AccountSubscription): Promise<void>` - Unsubscribe from account activity
+
+#### Events Published
+
+- `AccountActivityService:balanceUpdated` - Real-time balance changes
+- `AccountActivityService:transactionUpdated` - Transaction status updates
+- `AccountActivityService:statusChanged` - Chain/service status changes
+
+## Contributing
+
+Please follow MetaMask's [contribution guidelines](../../CONTRIBUTING.md) when submitting changes.
+
+### Development
+
+```bash
+# Install dependencies
+yarn install
+
+# Run tests
+yarn test
+
+# Build
+yarn build
+
+# Lint
+yarn lint
+```
+
+### Testing
+
+Run the test suite to ensure your changes don't break existing functionality:
+
+```bash
+yarn test
+```
```

Comment:
All of this is listed in the Contributing doc, does it need to be here? It also lists commands for working with the entire monorepo when working with just this one package might be more useful.

``` suggestion
```